test/unittest/memory/unittest_memory_pool.cpp

```diff
@@ -0,0 +1,544 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2021 Parichay Kapoor <pk.kapoor@samsung.com>
+ *
+ * @file unittest_memory_pool.cpp
+ * @date 11 August 2021
+ * @brief Memory Pool Test
+ * @see	https://github.com/nnstreamer/nntrainer
+ * @author Parichay Kapoor <pk.kapoor@samsung.com>
+ * @bug No known bugs except for NYI items
+ */
+
+#include <cstring>
+#include <random>
+#include <vector>
+
+#include <gtest/gtest.h>
+
+#include <basic_planner.h>
+#include <memory_planner_validate.h>
+#include <memory_pool.h>
+
+constexpr unsigned int MEM_BYTES = 128;
+constexpr unsigned int MEM_QUANT = 100;
+constexpr unsigned int INTERVAL_SIZE = 5;
+
+/**
+ * @brief creation and destruction
+ */
+TEST(MemoryPool, create_destroy) { EXPECT_NO_THROW(nntrainer::MemoryPool()); }
+
+/**
+ * @brief request 0 sized memory
+ */
+TEST(MemoryPool, request_mem_01_n) {
+  nntrainer::MemoryPool pool;
+
+  EXPECT_THROW(pool.requestMemory(0, 1, 2), std::invalid_argument);
+}
+
+/**
+ * @brief request memory when starts after it ends
+ */
+TEST(MemoryPool, request_mem_02_n) {
+  nntrainer::MemoryPool pool;
+
+  EXPECT_THROW(pool.requestMemory(1, 3, 2), std::invalid_argument);
+}
+
+/**
+ * @brief request memory with 0 valid time
+ */
+TEST(MemoryPool, request_mem_03_n) {
+  nntrainer::MemoryPool pool;
+
+  EXPECT_THROW(pool.requestMemory(1, 4, 4), std::invalid_argument);
```

Comment:
is the interval half-open?

Reply:
start time is inclusive and end time is exclusive (as noted in memory pool documentation).
As this this leads to zero time duration, the function call throws.

Reply:
👍